pallets/loans-ref/src/migrations.rs

```diff
@@ -0,0 +1,90 @@
+#[cfg(feature = "try-runtime")]
+use frame_support::ensure;
+use frame_support::{
+	pallet_prelude::ValueQuery,
+	storage, storage_alias,
+	traits::{Get, OnRuntimeUpgrade},
+	weights::Weight,
+	Blake2_128Concat,
+};
+#[cfg(feature = "try-runtime")]
+use sp_std::vec::Vec;
+
+use crate::*;
+
+mod v0 {
+	use super::*;
+
+	/// This storage comes from the previous pallet loans.
+	/// It is used as an indicator that the previous pallet loans still exists.
+	/// If this storage is not found, the nuking process is aborted.
+	#[storage_alias]
+	pub(crate) type NextLoanId<T: Config> =
+		StorageMap<Pallet<T>, Blake2_128Concat, PoolIdOf<T>, u128, ValueQuery>;
+}
+
+/// This migration nukes all storages from the pallet individually.
+pub struct NukeMigration<T>(sp_std::marker::PhantomData<T>);
+
+impl<T: Config> OnRuntimeUpgrade for NukeMigration<T> {
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+		ensure!(
+			contains_prefixed_key(&loan_prefix()),
+			"Pallet loans prefix doesn't exists"
+		);
+
+		ensure!(
+			v0::NextLoanId::<T>::iter_values().count() == 1,
+			"Pallet loans contains doesn't contain old data"
+		);
+
+		Ok(Vec::new())
+	}
+
+	fn on_runtime_upgrade() -> Weight {
+		if v0::NextLoanId::<T>::iter_values().count() == 1 {
+			let result = storage::unhashed::clear_prefix(&loan_prefix(), None, None);
+
+			match result.maybe_cursor {
+				None => log::info!("Loans: storage cleared successful"),
+				Some(_) => log::error!("Loans: storage not totally cleared"),
+			}
+
+			T::DbWeight::get().writes(result.unique.into())
+				+ T::DbWeight::get().reads(result.loops.into())
+		} else {
+			log::warn!("Loans: storage was already clear. This migration can be removed.");
+
+			Weight::zero()
```

Comment:
AFAIK we still perform a `read`

```suggestion
			T::DbWeight::get().reads(1)
```

Reply:
You're right!

Reply:
Following your hint, I've made a minor change for the sake of purism